### ISSUE TITLE
CI: [v1.11] Jenkins tests failing on 4.9, agent trying to pass BPF ISA set `v3` to `init.sh`

### ISSUE THREAD

**Issue by qmonnet** [cilium/cilium#24394]
### Test Name

Kernel 4.9, but does not seem related to one test or K8s version in particular:

```
Suite-k8s-1.19.K8sDatapathConfig MonitorAggregation Checks that monitor aggregation restricts notifications
Suite-k8s-1.21.K8sServicesTest Checks E/W loadbalancing (ClusterIP, NodePort from inside cluster, etc) Checks service on same node
```

### Failure Output

```shell
⚠️  Found "2023-03-14T18:45:29.040496972Z level=error msg=\"Command execution failed\" cmd=\"[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.1.190 fd04::12 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]\" error=\"exit status 1\" subsys=datapath-loader" in logs 1 times
⚠️  Found "2023-03-14T18:45:27.158158954Z level=error msg=\"Command execution failed\" cmd=\"[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.0.188 fd04::11 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]\" error=\"exit status 1\" subsys=datapath-loader" in logs 1 times
```

### Stack Trace

```shell
/home/jenkins/workspace/Cilium-PR-K8s-1.19-kernel-4.9/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:427
Found 2 k8s-app=cilium logs matching list of errors that must be investigated:
2023-03-14T18:45:29.040496972Z level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.1.190 fd04::12 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]" error="exit status 1" subsys=datapath-loader
2023-03-14T18:45:27.158158954Z level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.0.188 fd04::11 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]" error="exit status 1" subsys=datapath-loader
/home/jenkins/workspace/Cilium-PR-K8s-1.19-kernel-4.9/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:425
```

### Standard Output

```shell
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
No errors/warnings found in logs
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
No errors/warnings found in logs
⚠️  Found "2023-03-14T18:45:29.040496972Z level=error msg=\"Command execution failed\" cmd=\"[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.1.190 fd04::12 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]\" error=\"exit status 1\" subsys=datapath-loader" in logs 1 times
⚠️  Found "2023-03-14T18:45:27.158158954Z level=error msg=\"Command execution failed\" cmd=\"[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.0.188 fd04::11 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]\" error=\"exit status 1\" subsys=datapath-loader" in logs 1 times
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 2
⚠️  Number of "level=warning" in logs: 460
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
Top 5 errors/warnings:
Command execution failed
100:\tfrom all lookup local
+ true
+ '[' false = true ']'
++ awk '{print $2}'
Cilium pods: [cilium-9vdqs cilium-qht5n]
Netpols loaded: 
CiliumNetworkPolicies loaded: 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito::l3-policy-demo 
Endpoint Policy Enforcement:
Pod                          Ingress   Egress
testclient-z2tw7                       
testds-85gmn                           
testds-vmwpm                           
coredns-7c74c644b-49p9h                
test-k8s2-79ff876c9d-wxqv8             
testclient-qc75m                       
Cilium agent 'cilium-9vdqs': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 35 Failed 0
Cilium agent 'cilium-qht5n': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 27 Failed 0
```

### Standard Error

```shell
18:45:04 STEP: Running BeforeAll block for EntireTestsuite K8sDatapathConfig
18:45:04 STEP: Ensuring the namespace kube-system exists
18:45:04 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs")
18:45:04 STEP: WaitforPods(namespace="kube-system", filter="-l k8s-app=cilium-test-logs") => <nil>
18:45:04 STEP: Installing Cilium
18:45:05 STEP: Waiting for Cilium to become ready
18:45:54 STEP: Restarting unmanaged pods coredns-7c74c644b-st22j in namespace kube-system
18:46:03 STEP: Validating if Kubernetes DNS is deployed
18:46:03 STEP: Checking if deployment is ready
18:46:03 STEP: Kubernetes DNS is not ready: only 0 of 1 replicas are available
18:46:03 STEP: Restarting Kubernetes DNS (-l k8s-app=kube-dns)
18:46:03 STEP: Waiting for Kubernetes DNS to become operational
18:46:03 STEP: Checking if deployment is ready
18:46:03 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:04 STEP: Checking if deployment is ready
18:46:04 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:05 STEP: Checking if deployment is ready
18:46:05 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:06 STEP: Checking if deployment is ready
18:46:06 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:07 STEP: Checking if deployment is ready
18:46:07 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:08 STEP: Checking if deployment is ready
18:46:08 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:09 STEP: Checking if deployment is ready
18:46:09 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:10 STEP: Checking if deployment is ready
18:46:10 STEP: Kubernetes DNS is not ready yet: only 0 of 1 replicas are available
18:46:11 STEP: Checking if deployment is ready
18:46:11 STEP: Checking if kube-dns service is plumbed correctly
18:46:11 STEP: Checking if pods have identity
18:46:11 STEP: Checking if DNS can resolve
18:46:15 STEP: Kubernetes DNS is not ready yet: CiliumEndpoint does not exist
18:46:15 STEP: Checking if deployment is ready
18:46:15 STEP: Checking if kube-dns service is plumbed correctly
18:46:15 STEP: Checking if DNS can resolve
18:46:15 STEP: Checking if pods have identity
18:46:19 STEP: Kubernetes DNS is not ready yet: CiliumEndpoint does not exist
18:46:19 STEP: Checking if deployment is ready
18:46:19 STEP: Checking if kube-dns service is plumbed correctly
18:46:19 STEP: Checking if pods have identity
18:46:19 STEP: Checking if DNS can resolve
18:46:23 STEP: Validating Cilium Installation
18:46:23 STEP: Performing Cilium controllers preflight check
18:46:23 STEP: Performing Cilium health check
18:46:23 STEP: Performing Cilium status preflight check
18:46:23 STEP: Checking whether host EP regenerated
18:46:31 STEP: Performing Cilium service preflight check
18:46:31 STEP: Performing K8s service preflight check
18:46:37 STEP: Waiting for cilium-operator to be ready
18:46:37 STEP: WaitforPods(namespace="kube-system", filter="-l name=cilium-operator")
18:46:37 STEP: WaitforPods(namespace="kube-system", filter="-l name=cilium-operator") => <nil>
18:46:37 STEP: Making sure all endpoints are in ready state
18:46:40 STEP: Launching cilium monitor on "cilium-qht5n"
18:46:40 STEP: Creating namespace 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito
18:46:40 STEP: Deploying demo_ds.yaml in namespace 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito
18:46:40 STEP: Applying policy /home/jenkins/workspace/Cilium-PR-K8s-1.19-kernel-4.9/src/github.com/cilium/cilium/test/k8sT/manifests/l3-policy-demo.yaml
18:46:48 STEP: Waiting for 4m0s for 5 pods of deployment demo_ds.yaml to become ready
18:46:48 STEP: WaitforNPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="")
18:46:53 STEP: WaitforNPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="") => <nil>
18:46:53 STEP: Checking pod connectivity between nodes
18:46:53 STEP: WaitforPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="-l zgroup=testDSClient")
18:46:54 STEP: WaitforPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="-l zgroup=testDSClient") => <nil>
18:46:54 STEP: WaitforPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="-l zgroup=testDS")
18:46:54 STEP: WaitforPods(namespace="202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito", filter="-l zgroup=testDS") => <nil>
18:46:58 STEP: Checking that ICMP notifications in egress direction were observed
18:46:58 STEP: Checking that ICMP notifications in ingress direction were observed
18:46:58 STEP: Checking the set of TCP notifications received matches expectations
18:46:58 STEP: Looking for TCP notifications using the ephemeral port "39898"
=== Test Finished at 2023-03-14T18:46:58Z====
18:46:58 STEP: Running JustAfterEach block for EntireTestsuite K8sDatapathConfig
FAIL: Found 2 k8s-app=cilium logs matching list of errors that must be investigated:
2023-03-14T18:45:29.040496972Z level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.1.190 fd04::12 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]" error="exit status 1" subsys=datapath-loader
2023-03-14T18:45:27.158158954Z level=error msg="Command execution failed" cmd="[/var/lib/cilium/bpf/init.sh /var/lib/cilium/bpf /var/run/cilium/state 10.0.0.188 fd04::11 tunnel vxlan 8472 <nil> cilium_host cilium_net 1500 false false false /run/cilium/cgroupv2 /sys/fs/bpf false true v3 3 false true]" error="exit status 1" subsys=datapath-loader
===================== TEST FAILED =====================
18:46:59 STEP: Running AfterFailed block for EntireTestsuite K8sDatapathConfig
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE                                                         NAME                              READY   STATUS    RESTARTS   AGE    IP              NODE   NOMINATED NODE   READINESS GATES
	 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito   test-k8s2-79ff876c9d-wxqv8        2/2     Running   0          23s    10.0.1.216      k8s2   <none>           <none>
	 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito   testclient-qc75m                  1/1     Running   0          23s    10.0.0.140      k8s1   <none>           <none>
	 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito   testclient-z2tw7                  1/1     Running   0          23s    10.0.1.108      k8s2   <none>           <none>
	 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito   testds-85gmn                      2/2     Running   0          23s    10.0.0.42       k8s1   <none>           <none>
	 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito   testds-vmwpm                      2/2     Running   0          23s    10.0.1.140      k8s2   <none>           <none>
	 cilium-monitoring                                                 grafana-d69c97b9b-xcdm8           0/1     Running   0          28m    10.0.1.154      k8s2   <none>           <none>
	 cilium-monitoring                                                 prometheus-655fb888d7-4rnm9       1/1     Running   0          28m    10.0.0.199      k8s1   <none>           <none>
	 kube-system                                                       cilium-9vdqs                      1/1     Running   1          118s   192.168.56.12   k8s2   <none>           <none>
	 kube-system                                                       cilium-operator-b5dd6586c-8dxwk   1/1     Running   0          118s   192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       cilium-operator-b5dd6586c-mhqrh   1/1     Running   0          118s   192.168.56.12   k8s2   <none>           <none>
	 kube-system                                                       cilium-qht5n                      1/1     Running   1          118s   192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       coredns-7c74c644b-49p9h           1/1     Running   0          60s    10.0.1.110      k8s2   <none>           <none>
	 kube-system                                                       etcd-k8s1                         1/1     Running   0          38m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       kube-apiserver-k8s1               1/1     Running   2          38m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       kube-controller-manager-k8s1      1/1     Running   3          38m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       kube-proxy-9gpnz                  1/1     Running   0          37m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       kube-proxy-9pfbw                  1/1     Running   0          29m    192.168.56.12   k8s2   <none>           <none>
	 kube-system                                                       kube-scheduler-k8s1               1/1     Running   3          38m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       log-gatherer-kjj9f                1/1     Running   0          28m    192.168.56.11   k8s1   <none>           <none>
	 kube-system                                                       log-gatherer-v8jng                1/1     Running   0          28m    192.168.56.12   k8s2   <none>           <none>
	 kube-system                                                       registry-adder-t7gbb              1/1     Running   0          29m    192.168.56.12   k8s2   <none>           <none>
	 kube-system                                                       registry-adder-zhrf7              1/1     Running   0          29m    192.168.56.11   k8s1   <none>           <none>
	 
Stderr:
 	 

Fetching command output from pods [cilium-9vdqs cilium-qht5n]
cmd: kubectl exec -n kube-system cilium-9vdqs -c cilium-agent -- cilium status
Exitcode: 0 
Stdout:
 	 KVStore:                Ok   Disabled
	 Kubernetes:             Ok   1.19 (v1.19.16) [linux/amd64]
	 Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1beta1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
	 KubeProxyReplacement:   Disabled   
	 Host firewall:          Disabled
	 Cilium:                 Ok   1.11.14 (v1.11.14-1d8644c)
	 NodeMonitor:            Listening for events on 3 CPUs with 64x4096 of shared memory
	 Cilium health daemon:   Ok   
	 IPAM:                   IPv4: 6/254 allocated from 10.0.1.0/24, IPv6: 6/254 allocated from fd02::100/120
	 BandwidthManager:       Disabled
	 Host Routing:           Legacy
	 Masquerading:           IPTables [IPv4: Enabled, IPv6: Enabled]
	 Controller Status:      35/35 healthy
	 Proxy Status:           OK, ip 10.0.1.190, 0 redirects active on ports 10000-20000
	 Hubble:                 Ok   Current/Max Flows: 363/65535 (0.55%), Flows/s: 2.82   Metrics: Disabled
	 Encryption:             Disabled
	 Cluster health:         2/2 reachable   (2023-03-14T18:46:30Z)
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-9vdqs -c cilium-agent -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                                                                       IPv6        IPv4         STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                                             
	 21         Enabled            Disabled          32007      k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::1e5   10.0.1.140   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                    
	                                                            k8s:io.kubernetes.pod.namespace=202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito                                    
	                                                            k8s:zgroup=testDS                                                                                                                  
	 53         Disabled           Disabled          54644      k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::1b6   10.0.1.216   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                    
	                                                            k8s:io.kubernetes.pod.namespace=202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito                                    
	                                                            k8s:zgroup=test-k8s2                                                                                                               
	 665        Disabled           Disabled          1          k8s:cilium.io/ci-node=k8s2                                                                                                 ready   
	                                                            reserved:host                                                                                                                      
	 2541       Disabled           Disabled          4          reserved:health                                                                                   fd02::13c   10.0.1.242   ready   
	 3250       Disabled           Disabled          10508      k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::1c7   10.0.1.108   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                    
	                                                            k8s:io.kubernetes.pod.namespace=202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito                                    
	                                                            k8s:zgroup=testDSClient                                                                                                            
	 3972       Disabled           Disabled          3154       k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::155   10.0.1.110   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                                                    
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                                                        
	                                                            k8s:k8s-app=kube-dns                                                                                                               
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-qht5n -c cilium-agent -- cilium status
Exitcode: 0 
Stdout:
 	 KVStore:                Ok   Disabled
	 Kubernetes:             Ok   1.19 (v1.19.16) [linux/amd64]
	 Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1beta1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
	 KubeProxyReplacement:   Disabled   
	 Host firewall:          Disabled
	 Cilium:                 Ok   1.11.14 (v1.11.14-1d8644c)
	 NodeMonitor:            Listening for events on 3 CPUs with 64x4096 of shared memory
	 Cilium health daemon:   Ok   
	 IPAM:                   IPv4: 4/254 allocated from 10.0.0.0/24, IPv6: 4/254 allocated from fd02::/120
	 BandwidthManager:       Disabled
	 Host Routing:           Legacy
	 Masquerading:           IPTables [IPv4: Enabled, IPv6: Enabled]
	 Controller Status:      27/27 healthy
	 Proxy Status:           OK, ip 10.0.0.188, 0 redirects active on ports 10000-20000
	 Hubble:                 Ok   Current/Max Flows: 413/65535 (0.63%), Flows/s: 3.21   Metrics: Disabled
	 Encryption:             Disabled
	 Cluster health:         2/2 reachable   (2023-03-14T18:46:40Z)
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-qht5n -c cilium-agent -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                                                                       IPv6       IPv4         STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                                            
	 332        Disabled           Disabled          4          reserved:health                                                                                   fd02::a1   10.0.0.23    ready   
	 560        Disabled           Disabled          10508      k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::36   10.0.0.140   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                   
	                                                            k8s:io.kubernetes.pod.namespace=202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito                                   
	                                                            k8s:zgroup=testDSClient                                                                                                           
	 1220       Disabled           Disabled          1          k8s:cilium.io/ci-node=k8s1                                                                                                ready   
	                                                            k8s:node-role.kubernetes.io/master                                                                                                
	                                                            reserved:host                                                                                                                     
	 3499       Enabled            Disabled          32007      k8s:io.cilium.k8s.policy.cluster=default                                                          fd02::11   10.0.0.42    ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                                                   
	                                                            k8s:io.kubernetes.pod.namespace=202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito                                   
	                                                            k8s:zgroup=testDS                                                                                                                 
	 
Stderr:
 	 

===================== Exiting AfterFailed =====================
18:47:11 STEP: Running AfterEach for block EntireTestsuite K8sDatapathConfig
18:47:11 STEP: Deleting deployment demo_ds.yaml
18:47:11 STEP: Deleting namespace 202303141846k8sdatapathconfigmonitoraggregationchecksthatmonito
18:47:25 STEP: Running AfterEach for block EntireTestsuite
```

### Resources

Occurrence 1 (k8s-1.19)

- [Jenkins URL](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.19-kernel-4.9/2716/testReport/junit/Suite-k8s-1/19/K8sDatapathConfig_MonitorAggregation_Checks_that_monitor_aggregation_restricts_notifications/)
- [ZIP file](https://github.com/cilium/cilium/files/10983093/83815a32_K8sDatapathConfig_MonitorAggregation_Checks_that_monitor_aggregation_restricts_notifications.zip)

Occurrence 2 (k8s-1.21)

- [Jenkins URL](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/2486/testReport/junit/Suite-k8s-1/21/K8sServicesTest_Checks_E_W_loadbalancing__ClusterIP__NodePort_from_inside_cluster__etc__Checks_service_on_same_node/)
- [ZIP file](https://github.com/cilium/cilium/files/10983095/696fabf0_K8sServicesTest_Checks_E-W_loadbalancing_.ClusterIP._NodePort_from_inside_cluster._etc._Checks_service_on_same_node.zip)

Both spotted on https://github.com/cilium/cilium/pull/24368

### Anything else?

We noticed that bpftool's feature probing contains some mistakes on 4.9 (some recent types and their related helpers are marked as supported, but they're not). However, from the plain-text dump from `bpftool feature probe` in the logs, it looks like helper `bpf_redirect_neigh` for `sched_cls` programs is correctly probed as *not* being present on the system.

The agent uses the JSON output from bpftool, missing from the logs, but seemingly consistent with the plain output (from tests in Vagrant on 4.9 image used with cilium v1.11).

Then this issue happens, agent logs have warning from libbpf DUMP LOG like this: `unknown insn class 6`

**Comment by pchaigno** [comment]
@qmonnet Are you still looking into this? What's the next step?

**Comment by qmonnet** [comment]
Yes. I haven't been able to reproduce locally.

Next steps (for sometime next week):

- Add logs and create a PR, run Jenkins CI until we get more relevant logs.
- Look at Git history around the date where the first flakes started to happen - it's 1.11, we're only merging so many commits after all.

**Comment by nbusseneau** [comment]
I have hit this in #25011:
- https://jenkins.cilium.io/job/Cilium-PR-K8s-1.17-kernel-4.9/1315/testReport/junit/Suite-k8s-1/17/K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies/
  - [84d103f9_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip](https://github.com/cilium/cilium/files/11296842/84d103f9_K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies.zip)
- https://jenkins.cilium.io/job/Cilium-PR-K8s-1.19-kernel-4.9/2822/testReport/junit/Suite-k8s-1/19/K8sHubbleTest_Hubble_Observe_Test_L3_L4_Flow/
  - [e4841f60_K8sHubbleTest_Hubble_Observe_Test_L3-L4_Flow.zip](https://github.com/cilium/cilium/files/11296844/e4841f60_K8sHubbleTest_Hubble_Observe_Test_L3-L4_Flow.zip)
- https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/1819/testReport/junit/Suite-k8s-1/20/K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies/
  - [425c758e_K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies.zip](https://github.com/cilium/cilium/files/11296849/425c758e_K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies.zip)
- https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/2589/testReport/junit/Suite-k8s-1/21/K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies/
  - [9e66a052_K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies.zip](https://github.com/cilium/cilium/files/11296853/9e66a052_K8sPolicyTest_Basic_Test_checks_all_kind_of_Kubernetes_policies.zip)

**Comment by qmonnet** [comment]
The current hypothesis is that it could be a memlock/rlimit issue:

- This would explain why all map types are reported as unavailable when we get the unexpected detection results.

- This would explain why bpftool believes the helper is available: To find whether a BPF helper is available for a given program type, libbpf tries to load a program with two instructions: `call <helper>; return 0;`. It does not care whether the load succeeds; instead it greps for a specific string in the verifier logs. If the string is present, the helper is unavailable. If the string is not there, the helper is presumed available. But if we have a memlock issue, then maybe no verifier logs are produced because we get rejected even before the verifier, in which case we can't have that string in the logs, and report the helper as available (Should bpftool report the helper as unavailable on `EPERM`? It's tricky, because `EPERM` can also be produced in other situations where the helper is actually available).

- But then the memlock issue is not supposed to happen in the first place, given that we raise  the rlimit to infinity in bpftool before running the probes.

As for the last bullet point, this is reminiscent of #20288, where we were hitting memlock issues due to a kernel bug. The symptoms were different (because we'd hit it when loading the agent programs, not when probing for features), but the bug could well be the same. We fixed it and backported to the 4.9 boxes used in the CI for v1.12, but not for v1.11. In #25247, I'll try to run the CI a few times after bumping the kernel version in the Vagrant box, to include the fix for this bug. Let's see how it goes.

**Comment by qmonnet** [comment]
Hopefully fixed with https://github.com/cilium/cilium/pull/25247. Please reopen and ping me if you spot new occurrences.
